core/class/new_spec.rb

```diff
@@ -71,6 +71,7 @@ def message2; "hello"; end
     obj = mock("Class.new metaclass")
     meta = obj.singleton_class
     -> { Class.new meta }.should raise_error(TypeError)
+    -> { class TestClass < meta; end }.should raise_error(TypeError)
```

Comment:
This file is for `Class.new` specs, and as you've seen the `class` keyword is not quite the same as `Class.new` at least for the implementation. I'd suggest to move it to https://github.com/ruby/spec/blob/master/language/class_spec.rb.

Reply:
Didn't saw that one, looks like there already is test for this case https://github.com/ruby/spec/blob/master/language/class_spec.rb#L151
So this one is not needed and I'll move other one to there.